docs/DevGuide/Profiling.md

```diff
@@ -0,0 +1,187 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Profiling {#profiling}
+
+There are a number of tools available for profiling, each with their own
+strengths and weaknesses. This makes it difficult to recommend one "right" way
+of analyzing performance using profilers. Instead, one should use a combination
+of the tools to discover and eliminate performance bottle necks. Common
+profilers are Charm++ Projections (tracing-based), HPCToolkit (sampling-based,
+very versatile), Linux perf (sampling-based, command line only), Intel VTune
+(sampling-based, works well on Intel hardware), and AMD uProf (similar to Intel
+VTune).
+
+## Profiling with HPCToolkit {#profiling_with_hpctoolkit}
+
+Follow the HPCToolkit installation instructions at
+[hpctoolkit.org](http://hpctoolkit.org). The Spack
+installation seems to work well. Once installed, compile your executable in
+Release mode with `-D ENABLE_PROFILING=ON -D DEBUG_SYMBOLS=ON` since otherwise
+you won't be able to get call stacks and source analysis. Using `-D
+BUILD_SHARED_LIBS=ON` is recommended since it makes HPCToolkit a lot easier to
+use. You must also use the system allocator, `-D MEMORY_ALLOCATOR=SYSTEM`. We
+will work from the build directory and perform all runs and performance analysis
+there.
+
+First run HPCToolkit as:
+```
+hpcrun -t --event CYCLES@f200 ./bin/EXEC --input-file ./Input.yaml +p1
+```
+We will profile on one core, but you can profile on multiple cores as well as
+multiple nodes if using MPI as the Charm++ backend. This will generate a
+`hpctoolkit-EXEC-measurements` directory. Run
+```
+hpcstruct -jN ./hpctoolkit-EXEC-measurements
+```
+where `N` is the number of cores to run on. This will generate an the mapping to
+line numbers, etc. in the measurements directory.
+
+\warning Skipping the `hpcstruct` step will make `hprprof` below run extremely
+slowly.
+
+Once the run is complete, run
+```
+hpcprof -I /path/to/spectre/src/+ hpctoolkit-EXEC-measurements
```

Comment:
Can you add a note here that the `+` is literally supposed to be the `+` symbol. I was confused when I saw that and thought I had to do something else.